### ISSUE TITLE
'eslint --init' installs wrong versions of depencies for AirBnB

### ISSUE THREAD

**Issue by carlin-q-scott** [eslint/eslint#7338]
**Tell us about your environment**
Windows 10
- **ESLint Version:** 3.7.1
- **Node Version:** 5.3.0
- **npm Version:** 3.3.12

**What parser (default, Babel-ESLint, etc.) are you using?**
N/A
**Please show your full configuration:**
AirBnB style guide
**What did you do? Please include the actual source code causing the issue.**

```
npm init
Use a popular style guide
Airbnb
JSON
```

**What did you expect to happen?**
I expected the node modules to install correctly.
**What actually happened? Please include the actual, raw output from ESLint.**
Eslint failed to correctly install dependencies as seen at the bottom of this output:

```
C:\SWDev\job-frontend>eslint --init                                                                                                                                                          
? How would you like to configure ESLint? Use a popular style guide                                                                                                                          
? Which style guide do you want to follow? Airbnb                                                                                                                                            
? What format do you want your config file to be in? JSON                                                                                                                                    
Installing eslint-plugin-react, eslint-plugin-jsx-a11y, eslint-plugin-import, eslint-config-airbnb                                                                                           
npm WARN install Couldn't install optional dependency: Unsupported                                                                                                                           
job-frontend@0.0.0 C:\SWDev\job-frontend                                                                                                                                                     
├─┬ eslint@3.7.1                                                                                                                                                                             
│ └── doctrine@1.3.0                                                                                                                                                                         
├─┬ eslint-config-airbnb@12.0.0                                                                                                                                                              
│ ├── eslint-config-airbnb-base@8.0.0                                                                                                                                                        
│ └── UNMET PEER DEPENDENCY eslint-plugin-import@^1.16.0                                                                                                                                     
├─┬ UNMET PEER DEPENDENCY eslint-plugin-import@2.0.0                                                                                                                                         
│ ├── contains-path@0.1.0                                                                                                                                                                    
│ ├── eslint-import-resolver-node@0.2.3   
│ ├─┬ eslint-module-utils@1.0.0                                                                                                                                                              
│ │ └── pkg-dir@1.0.0                                                                                                                                                                        
│ ├── lodash.cond@4.5.2                                                                                                                                                                      
│ └── pkg-up@1.0.0                                                                                                                                                                           
├─┬ eslint-plugin-jsx-a11y@2.2.3                                                                                                                                                             
│ └── damerau-levenshtein@1.0.3                                                                                                                                                              
└── eslint-plugin-react@6.4.1                                                                                                                                                                

npm WARN EPEERINVALID eslint-config-airbnb@12.0.0 requires a peer of eslint-plugin-import@^1.16.0 but none was installed.                                                                    
npm WARN EPEERINVALID eslint-config-airbnb-base@8.0.0 requires a peer of eslint-plugin-import@^1.16.0 but none was installed.                                                                
npm WARN EPACKAGEJSON job-frontend@0.0.0 No license field.
```

**Workaround**
The AirBnB style guide for standard javascript is actually eslint-config-airbnb-base, the one used by the wizard is for react. So I swapped out that package and chose the versions of the decencies that would work (with either react or standard AirBnB).

Initialized package.json:

```
    "eslint-config-airbnb": "^12.0.0",
    "eslint-plugin-import": "^2.0.0",
    "eslint-plugin-jsx-a11y": "^2.2.3",
```

Updated package.json:

```
    "eslint-config-airbnb-base": "^8.0.0",
    "eslint-plugin-import": "^1.16.0",
    "eslint-plugin-jsx-a11y": "^2.2.2",
```

**Comment by platinumazure** [comment]
Ping @ljharb, can you help?

**Comment by ljharb** [comment]
Please use the bash command in the readme of `eslint-config-airbnb` to install all the proper peer deps - reproduced here (i'm pretty sure i mentioned it in another issue as well):
```sh
(
  export PKG=eslint-config-airbnb;
  npm info "$PKG" peerDependencies --json | command sed 's/[\{\},]//g ; s/: /@/g' | xargs npm install --save-dev "$PKG"
)
```

I highly recommend reusing that command for anything you install that might have peer deps.

**Comment by ilyavolodin** [comment]
@ljharb We can't use this command, since it wouldn't work on Windows. I'm not even sure it would work on OSX without some additional installations.

**Comment by vitorbal** [comment]
@ilyavolodin @ljharb how about something like this:
```javascript
const npmOutput = shell.exec(`npm info ${packageName} peerDependencies --json`, {silent: "true"}).stdout;
const peerDependencies = JSON.parse(npmOutput);
```
Would it be safe to assume that an npm executable is present? Otherwise we could still fall back to a hardcoded version..

(I was actually hacking a bit on this the other day, but paused it for now to pursue some JSCS milestone issues)

**Comment by ljharb** [comment]
@ilyavolodin it absolutely works on stock OS X as well as any POSIX system with npm installed. You're right about Windows, but you can reimplement that command in JS pretty trivially, as shown above.

**Comment by ilyavolodin** [comment]
Ok, I have no objections to doing it the way @vitorbal showed. To answer your question, yes, we should assume that npm executable is present.

**Comment by miqmago** [comment]
Is this also related with https://github.com/airbnb/javascript/issues/1112?

**Comment by ljharb** [comment]
@miqmago yes, that's the current cause. however, because eslint configs need to use peerDeps, any time a breaking change peerDep of the config is released, this problem will re-present itself, unless `--init` explicitly installs the peer deps that the intended config version depends on.

**Comment by platinumazure** [comment]
I've added "help wanted" label to this issue because this is probably something that will really help airbnb users (as well as standard/xo), but the core team doesn't have time to look at this right now.

**Comment by ljharb** [comment]
I started to work on this, but am blocked on https://github.com/npm/node-semver/issues/178 - ie, comparing the two semver ranges of a peerDep from the user's package.json, and the target eslint-config-foo's package.json.

**Comment by michaelmcmillan** [comment]
This is so broken it's not even funny.

**Comment by tirithen** [comment]
This has been broken for soo long now, are there any other close match standards that just works? Eslint seemed nice but it's split out in so many packages that this really is becoming a pain to setup.

**Comment by kaicataldo** [comment]
I'm a little bit confused as to if this is a confirmed bug or not. @platinumazure Can you confirm by any chance, since you put the "help wanted" label on it?

**Comment by platinumazure** [comment]
I'm not sure if it's a bug or an enhancement, but I believe we had
consensus to do something like `npm view eslint-plugin-airbnb
peerDependencies` and install the expected versions of peer dependencies. I
threw on "help wanted" because it seemed like it wasn't going to be a
priority for the core team due to JSCS compatibility.

**Comment by ljharb** [comment]
It's definitely a bug, due to the way eslint currently handles peerDependencies or shared configs - I've made progress locally per https://github.com/eslint/eslint/issues/7338#issuecomment-261487207, but am blocked as described.

**Comment by cloud-github** [comment]
`npm info "eslint-config-airbnb-base@latest" peerDependencies`
`npm install -g install-peerdeps`
`install-peerdeps --dev eslint-config-airbnb-base`

Add "extends": "airbnb-base" to .eslintrc   file. 
Finally, SUCCESS eslint-config-airbnb-base and its peerDeps were installed successfully.
